### PR TITLE
add partition key option to from_rows_table

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/matrixtable.py [hunk 1]
```diff
@@ -2725,12 +2725,21 @@ def from_rows_table(cls, table: Table) -> 'MatrixTable':
         ----------
         table : :class:`.Table`
             The table to be converted.
+        partition_key : :obj:`str` or :obj:`list` of :obj:`str`
+            Partition key(s).
```
Comment (jbloom22) [inline python/hail/matrixtable.py]: I think key should still be singular even when it has multiple fields. How about `Partition key field(s).`?

File: python/hail/matrixtable.py [hunk 2]
```diff
@@ -2725,12 +2725,21 @@ def from_rows_table(cls, table: Table) -> 'MatrixTable':
...
         Returns
         -------
         :class:`.MatrixTable`
         """
-        jmt = scala_object(Env.hail().variant, 'MatrixTable').fromRowsTable(table._jt)
+        if partition_key is not None:
+            if isinstance(partition_key, str):
+                partition_key = [partition_key]
+            if len(partition_key) == 0:
+                raise ValueError('partition_key may not be empty')
```
Comment (jbloom22) [inline python/hail/matrixtable.py]: 'partition_key must be non-empty' or 'partition_key must not be empty'

File: python/hail/matrixtable.py [hunk 3]
```diff
@@ -2725,12 +2725,21 @@ def from_rows_table(cls, table: Table) -> 'MatrixTable':
...
+            elif list(table.key)[:len(partition_key)] != partition_key:
+                raise ValueError('partition_key must be a prefix of table key')
```
Comment (jbloom22) [inline python/hail/matrixtable.py]: Yet the docs say the table must be keyed by a single field? Also, you should document such requirements.

File: python/hail/tests/test_api.py [hunk 4]
```diff
@@ -783,11 +783,8 @@ def test_filter_partitions(self):
     def test_from_rows_table(self):
         ds = hl.import_vcf(resource('sample.vcf'))
         rt = ds.rows()
-        rm = hl.MatrixTable.from_rows_table(rt)
-        # would be nice to compare rm to ds.drop_cols(), but the latter
-        # preserves the col, entry types
-        self.assertEqual(rm.count_cols(), 0)
-        self.assertTrue(rm.rows()._same(rt))
+        rm = hl.MatrixTable.from_rows_table(rt, partition_key='locus')
```
Comment (jbloom22) [inline python/hail/tests/test_api.py]: suggest renaming `ds` => `mt` and `rm` => `rmt`

Comment (tpoterba) [reply python/hail/tests/test_api.py]: This wasn't something I changed.  I just made the test better

File: python/hail/matrixtable.py (hunk 3 above)
Comment (tpoterba) [reply python/hail/matrixtable.py]: This is an experimental method that will be removed as soon as we have partitioned tables (a few weeks probably). Is it worth the hassle? I'll make these changes if you think so

Comment (jbloom22) [reply python/hail/matrixtable.py]: I still think the docs should reflect the current behavior if doing so is not onerous. Here you just need to change/add a few words.

File: python/hail/matrixtable.py [hunk 5]
```diff
@@ -2726,7 +2725,7 @@ def from_rows_table(cls, table: Table, partition_key: Optional[Union[str, List[s
         table : :class:`.Table`
             The table to be converted.
         partition_key : :obj:`str` or :obj:`list` of :obj:`str`
-            Partition key(s).
+            Partition key field(s).
```
Comment (jbloom22) [inline python/hail/matrixtable.py]: Just add "Must be a prefix of table key fields."

**Pull request merged**